### DISCUSSION TITLE
Could Nidhogg be bundled with a distributed software (and is it reliable enough to work on the vast majority of systems without causing any form of damage to the device/os)?

### DISCUSSION THREAD

**Discussion by ItsCubeTime** [Idov31/Nidhogg#47] [Q&A]
Howdy hey!

I'm developing an alternative to Microsoft Parental Controls and [Freedom](https://freedom.to/) that I'm hoping to eventually distribute to a larger audience. As a part of the softwares features, I need it to be able to protect itself from being tampered with while its running, more specifically:
* The user must not be able to terminate the program in a way that could prevent it from starting up again - even if the user has administrative priviligues. *So far, Ive written my program in such a way that it can resume operations even if the user were to terminate it, however aside from using something like Windows Task Scheduler I currently dont have a reliable way of making the program fire back up again if all processes were to get killed.*
* The program needs to be able to remember settings previously set by the user - and prevent those settings from being altered later on without relying on password based encryption.
* The program should start with Windows (this one I suppose is fairly straight forward if I were to just make my application add a shortcut to shell:startup , but that wouldn't work if the user were to try and start the computer in safe mode).

Hence the question: Would Nidhogg be fit to be bundled in a distributed software? If not, are there any other tools that you know of that would better fit my needs? I should also add  that I currently don't  have a driver signing certificate.

Im targeting Windows 11 first and foremost, Windows 10 being secondary.

Thank you - and sorry for invading your inbox with this.

**Comment by Idov31** [comment]
Hey @ItsCubeTime ,
Nidhogg is not suitable for this as it also contains other capabilities that can be abused in a bad way. Moreover, to use Nidhogg you will have to use either vulnerability or sign it with a legitimate signature or else the driver will not be loaded.

To achieve what you want you can install your program as a service with high privileges that runs automatically, restrict access to your program's folder and store your configuration file there.
